src/client/datascience/jupyter/kernels/helpers.ts

```diff
@@ -278,3 +281,35 @@ export function isLocalLaunch(configuration: IConfigurationService) {
 
     return false;
 }
+
+export async function sendTelemetryForPythonKernelExecutable(
+    notebook: INotebook,
+    file: string,
+    kernelConnection: KernelConnectionMetadata
+) {
+    if (!kernelConnection.interpreter || !isPythonKernelConnection(kernelConnection)) {
+        return;
+    }
+    if (kernelConnection.kind !== 'startUsingKernelSpec' && kernelConnection.kind !== 'startUsingPythonInterpreter') {
+        return;
+    }
+    try {
+        const cells = await notebook.execute('import sys\nprint(sys.executable)', file, 0, uuid(), undefined, true);
+        if (cells.length === 0 || !Array.isArray(cells[0].data.outputs) || cells[0].data.outputs.length === 0) {
+            return;
+        }
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const output: nbformat.IStream = cells[0].data.outputs[0] as any;
+        if (output.name !== 'stdout' && output.output_type !== 'stream') {
+            return;
+        }
+        const sysExecutable = concatMultilineString(output.text).trim().toLowerCase();
+        const match = kernelConnection.interpreter.path.toLowerCase() === sysExecutable;
```

Comment:
We don't care about case sensitivty for telemetry, 99.9% of the time its unlikely users will have two interpreters with a difference only in the case of the environment name.
I don't think conda would support that, & only applies to venvs, hence don't see the point in supporting that for telemetry.